src/main/scala/com/github/sbt/git/ConsoleGitRunner.scala

```diff
@@ -17,14 +18,14 @@ object ConsoleGitRunner extends GitRunner {
 
   // in order to enable colors we trick git into thinking we're a pager, because it already knows we're not a tty
   val colorSupport: Seq[(String, String)] =
-    if(ConsoleLogger.formatEnabled) Seq("GIT_PAGER_IN_USE" -> "1")
+    if(Terminal.console.isAnsiSupported) Seq("GIT_PAGER_IN_USE" -> "1")
```

Comment:
`Terminal.console` can throw an `IllegalStateException` [here](https://github.com/sbt/sbt/blob/8ead89691f9466cba698c989b4f4573897203f48/internal/util-logging/src/main/scala/sbt/internal/util/Terminal.scala#L847) so it should be wrapped with `try catch`.

Reply:
I used monadic `Try`, lemme know if you prefer `try/catch` for some reason. There's a few other places where `Try` is used in the plugin, but it's certainly not uniform.